### PR TITLE
Update pre commit hooks

### PR DESCRIPTION
**Pull request by jl-wynen**
This is based on #81

Updates hook versions and replaces the root level file with a minimal setup that works with the template repo. The other hooks run into problems because they can't parse pyproject.toml.jinja.

### CODE REVIEW

**Review by SimonHeybrock — COMMENTED** [review]
Should we add a static testing step in CI using those minimal hooks?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: BSD-3-Clause
```
Comment (SimonHeybrock) [inline .github/workflows/ci.yml:1]: Merge with `copier-ci.yml`?

Comment (jl-wynen) [reply .github/workflows/ci.yml:1]: I removed copier-ci in favour of this file

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
-name: CopierCI
+name: CI
 
 on:
   push:
     branches:
       - main
+      - release
```
Comment (YooSunYoung) [inline .github/workflows/ci.yml:10]: Will we ever use release on this...?

Comment (jl-wynen) [reply .github/workflows/ci.yml:10]: Probably not

**Pull request merged**